core/src/config.rs

```diff
@@ -28,82 +27,120 @@ impl<'de> Deserialize<'de> for MacAddr {
     where
         D: Deserializer<'de>,
     {
-        let s = <String>::deserialize(deserializer)?;
+        let s = String::deserialize(deserializer)?;
         MacAddr::from_str(&s).map_err(de::Error::custom)
     }
 }
 
 // make `Ipv4Cidr` serde deserializable.
 impl<'de> Deserialize<'de> for Ipv4Cidr {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
     where
         D: Deserializer<'de>,
     {
-        let s = <String>::deserialize(deserializer)?;
+        let s = String::deserialize(deserializer)?;
         Ipv4Cidr::from_str(&s).map_err(de::Error::custom)
     }
 }
 
 // make `Ipv6Cidr` serde deserializable.
 impl<'de> Deserialize<'de> for Ipv6Cidr {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
     where
         D: Deserializer<'de>,
     {
-        let s = <String>::deserialize(deserializer)?;
+        let s = String::deserialize(deserializer)?;
         Ipv6Cidr::from_str(&s).map_err(de::Error::custom)
     }
 }
 
-/// Runtime settings.
+/// Deserializes a duration from seconds expressed as `u64`.
+pub fn duration_from_secs<'de, D>(deserializer: D) -> std::result::Result<Duration, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let secs = u64::deserialize(deserializer)?;
+    Ok(Duration::from_secs(secs))
+}
```

Comment:
there's no magic that causes this to be called for any default deserialize of `Duration`, right? I see that we're explicitly specifying it down below, which is good.

Reply:
nope. no magic unfortunately. has to be explicitly referenced by the `serde` attribute.